tests/next_tests/unit_tests/errors_tests/test_formatting.py

```diff
@@ -0,0 +1,74 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import re
+
+import pytest
+
+from gt4py.eve import SourceLocation
+from gt4py.next.errors import DSLError
+from gt4py.next.errors.formatting import format_compilation_error
+
+
+@pytest.fixture
+def message():
+    return "a message"
+
+
+@pytest.fixture
+def location():
+    return SourceLocation("/source/file.py", 5, 2, end_line=5, end_column=9)
+
+
+@pytest.fixture
+def tb():
+    try:
+        raise Exception()
+    except Exception as ex:
+        return ex.__traceback__
+
+
+@pytest.fixture
+def type_():
+    return DSLError
+
+
+@pytest.fixture
+def qualname(type_):
+    return f"{type_.__module__}.{type_.__name__}"
+
+
+def test_format(type_, qualname, message):
+    pattern = f"{qualname}: {message}"
+    s = "\n".join(format_compilation_error(type_, message, None, None, None))
+    assert re.match(pattern, s)
+
+
+def test_format_loc(type_, qualname, message, location):
+    pattern = "Source location.*\\n" '  File "/source.*".*\\n' f"{qualname}: {message}"
```

Comment:
I think I wanted to split it into multiple lines so that it looks more similar to the actual text it represents, but the formatter was smart and put it all on one line.

I created separate variables for each line for now and joined them with `r"\n"`. The best alternative in my opinion would be to use a single-line multi-line raw f-string literal ( ̇r"""regex here""" ̇) so that quotes can be written easily.

Reply:
> The best alternative in my opinion would be to use a single-line multi-line raw f-string literal ( ̇r"""regex here""" ̇)

I guess you mean `ḟr"""regex here"""` or `̇fr'''regex here'''`. Both of them, and also using separate variables look good to me. Pick the one you like the most.

Reply:
Yeah, that's what I meant. I'll keep them as separate variables for now, we can evolve it in the future in either direction as requirements change.